### PR TITLE
Reduce the number of suppressions for internal access 

### PR DESCRIPTION
**Pull request by qwwdfsad**
While backporting K2 commits, I've spotted one related to `internal` declarations.

Instead of backporting it, I decided to tide it up in the face of https://youtrack.jetbrains.com/issue/KT-60866/Phase-out-usages-of-SuppressINVISIBLEREFERENCE-INVISIBLEMEMBER-in-libraries

### CODE REVIEW

File: core/common/src/serializers/DateTimeUnitSerializers.kt [hunk 1]
```diff
@@ -48,12 +47,12 @@ public object TimeBasedDateTimeUnitSerializer: KSerializer<DateTimeUnit.TimeBase
...
                         }
                         CompositeDecoder.DECODE_DONE -> break@loop // https://youtrack.jetbrains.com/issue/KT-42262
-                        else -> throw UnknownFieldException(elementIndex)
+                        else -> throwUnknownIndexException(elementIndex)
                     }
                 }
             }
         }
-        if (!seen) throw MissingFieldException("nanoseconds")
+        if (!seen) throw MissingFieldException(missingField = "nanoseconds", serialName = descriptor.serialName)
```
Comment (qwwdfsad) [inline core/common/src/serializers/DateTimeUnitSerializers.kt:55]: Explicit parameter names so one can be sure `missingField` and `serialName` (both are strings) are in the right order

File: core/common/src/serializers/DayOfWeekSerializers.kt [hunk 2]
```diff
@@ -16,14 +16,7 @@ import kotlinx.serialization.internal.*
...
-
-    override val descriptor: SerialDescriptor
-        get() = impl.descriptor
-
-    override fun deserialize(decoder: Decoder): DayOfWeek = impl.deserialize(decoder)
-
-    override fun serialize(encoder: Encoder, value: DayOfWeek): Unit = impl.serialize(encoder, value)
-}
+public object DayOfWeekSerializer : KSerializer<DayOfWeek> by createEnumSerializer<DayOfWeek>(
+    "kotlinx.datetime.serializers.DayOfWeek",
```
Comment (qwwdfsad) [inline core/common/src/serializers/DayOfWeekSerializers.kt:20]: Note that it is technically a breaking change: it provide FQ name as serial name (as we do and recommend doing in the serialization) and also replaces `Month` (which is a copy-paste error I believe)

**Comment by ilya-g** [comment]
Please exclude InlineOnly annotation removal, there's no problems with it in K2, as I was told in KT-60866

**Comment by qwwdfsad** [comment]
Done

File: core/common/src/serializers/DateTimeUnitSerializers.kt [hunk 3]
```diff
@@ -98,12 +96,12 @@ public object DayBasedDateTimeUnitSerializer: KSerializer<DateTimeUnit.DayBased>
...
                         }
                         CompositeDecoder.DECODE_DONE -> break@loop // https://youtrack.jetbrains.com/issue/KT-42262
-                        else -> throw UnknownFieldException(elementIndex)
+                        else -> throwUnknownIndexException(elementIndex)
                     }
                 }
             }
         }
-        if (!seen) throw MissingFieldException("days")
+        if (!seen) throw MissingFieldException(missingField = "days", serialName = TimeBasedDateTimeUnitSerializer.descriptor.serialName)
```
Comment (ilya-g) [inline core/common/src/serializers/DateTimeUnitSerializers.kt:104]: This looks suspicious. Is it intended to reference the descriptor of another type?

File: core/common/src/serializers/DayOfWeekSerializers.kt (hunk 2 above)
Comment (ilya-g) [reply core/common/src/serializers/DayOfWeekSerializers.kt:20]: By what principle this FQ name is built? Is it the name of the serialized type?

**Review by ilya-g — CHANGES_REQUESTED** [review]

File: core/common/src/serializers/DateTimeUnitSerializers.kt (hunk 3 above)
Comment (qwwdfsad) [reply core/common/src/serializers/DateTimeUnitSerializers.kt:104]: No, it's not. Thanks!

File: core/common/src/serializers/DayOfWeekSerializers.kt (hunk 2 above)
Comment (qwwdfsad) [reply core/common/src/serializers/DayOfWeekSerializers.kt:20]: According to the documentation:

>For generated serializers, serial name is equal to the corresponding class's fully-qualified name or, if overridden, SerialName. Custom serializers should provide a unique serial name that identify both the serializable class and the serializer itself, ignoring type arguments, if they are present.

We consider this one as "the default generated serializer" in fact, thus FQN of the serialized type.
I also opened https://github.com/Kotlin/kotlinx.serialization/pull/2581 to make it clear

**Comment by qwwdfsad** [comment]
@ilya-g sorry for the delay, it got lost from my radar. PTAL

File: core/common/src/serializers/DayOfWeekSerializers.kt (hunk 2 above)
Comment (ilya-g) [reply core/common/src/serializers/DayOfWeekSerializers.kt:20]: "FQN of the serialized type" - ok, then it should be changed as there is no type with this FQN :) 
Perhaps to "kotlinx.datetime.DayOfWeek"

Comment (qwwdfsad) [reply core/common/src/serializers/DayOfWeekSerializers.kt]: Changed it consistently in every serializer, PTAL

**Review by ilya-g — APPROVED** [review]

**Pull request merged**